File: Bubble sort/optimized_bubble_sort.py
def bubble_sort(array) :

    for x in range(len(array)-1) :
        flag = 0

        for y in range(len(array)-1-x) :

            if array[y] > array[y+1] :
                temp = array[y]
                array[y] = array[y+1]
                array[y+1] = temp
                flag = 1

        if flag == 0 :
            break
        
    return array

array = [1,2,3,4,5,6]
sorted_array = bubble_sort(array)
print("Array after sorting is ",sorted_array)

File: Bubble sort/optimized_bubble_sort.cpp
#include <stdio.h>
#include<stdlib.h>

void Bubble(int A[],int n)
{
int i,j,temp,flag;

for(i=0;i<n-1;i++)
{
flag = 0;

for(j=0;j<n-i-1;j++)
{

if(A[j]>A[j+1])
{
temp = A[j];
A[j] = A[j+1];
A[j+1] = temp;
flag = 1;
}
}

if(flag == 0 )
{
    break;
}
}
}

int main()
{
int A[]={1,2,3,4,5,6},i;
Bubble(A,sizeof(A)/sizeof(int));

printf("Array after sorting is [ ");
for(i=0;i<sizeof(A)/sizeof(int);i++)

printf("%d ",A[i]);
printf("]\n");

return 0;
}